controllers/cheuser/controller.go

```diff
@@ -0,0 +1,336 @@
+package cheuser
+
+import (
+	"context"
+
+	"github.com/devfile/devworkspace-operator/pkg/constants"
+	"github.com/devfile/devworkspace-operator/pkg/infrastructure"
+	org "github.com/eclipse-che/che-operator/api"
+	v1 "github.com/eclipse-che/che-operator/api/v1"
+	"github.com/eclipse-che/che-operator/api/v2alpha1"
+	"github.com/eclipse-che/che-operator/controllers/che"
+	"github.com/eclipse-che/che-operator/controllers/devworkspace"
+	"github.com/eclipse-che/che-operator/controllers/devworkspace/defaults"
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	projectv1 "github.com/openshift/api/project/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	userSettingsComponentLabelValue = "user-settings"
+)
+
+type CheUserNamespaceReconciler struct {
+	client         client.Client
+	scheme         *runtime.Scheme
+	namespaceCache namespaceCache
+}
+
+var _ reconcile.Reconciler = (*CheUserNamespaceReconciler)(nil)
+
+var (
+	configMapDiffOpts cmp.Option = cmp.Options{
+		cmpopts.IgnoreFields(corev1.ConfigMap{}, "TypeMeta", "ObjectMeta"),
+	}
+	secretDiffOpts cmp.Option = cmp.Options{
+		cmpopts.IgnoreFields(corev1.Secret{}, "TypeMeta", "ObjectMeta"),
+	}
+)
+
+func NewReconciler() *CheUserNamespaceReconciler {
+	return &CheUserNamespaceReconciler{namespaceCache: *NewNamespaceCache()}
+}
+
+func (r *CheUserNamespaceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.scheme = mgr.GetScheme()
+	r.client = mgr.GetClient()
+	r.namespaceCache.client = r.client
+
+	var obj runtime.Object
+	if infrastructure.IsOpenShift() {
+		obj = &projectv1.Project{}
+	} else {
+		obj = &corev1.Namespace{}
+	}
+
+	ctx := context.Background()
+	bld := ctrl.NewControllerManagedBy(mgr).
+		For(obj).
+		Watches(&source.Kind{Type: &corev1.Secret{}}, r.watchRulesForSecrets(ctx)).
+		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, r.watchRulesForConfigMaps(ctx)).
+		Watches(&source.Kind{Type: &v1.CheCluster{}}, r.triggerAllNamespaces(ctx))
+
+	return bld.Complete(r)
+}
+
+func (r *CheUserNamespaceReconciler) watchRulesForSecrets(ctx context.Context) handler.EventHandler {
+	return &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(mo handler.MapObject) []reconcile.Request {
+			if isLabeledAsUserSettings(mo.Meta) && r.isInManagedNamespace(ctx, mo.Meta) {
+				return asReconcileRequestForNamespace(mo.Meta)
+			} else if true {
```

Comment:
Sorry, maybe simple `else {` will be enough ?

Reply:
huh, where did that come from?  🤭